modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -149,18 +152,29 @@ public void clusterChanged(ClusterChangedEvent event) {
     }
 
     private void startTask(Runnable onFailure) {
-        persistentTasksService.sendStartRequest(
-            GEOIP_DOWNLOADER,
-            GEOIP_DOWNLOADER,
-            new GeoIpTaskParams(),
-            ActionListener.wrap(r -> logger.debug("Started geoip downloader task"), e -> {
-                Throwable t = e instanceof RemoteTransportException ? e.getCause() : e;
-                if (t instanceof ResourceAlreadyExistsException == false) {
-                    logger.error("failed to create geoip downloader task", e);
-                    onFailure.run();
-                }
-            })
-        );
+        boolean canStartTask;
+        try {
+            canStartTask = taskSemaphore.tryAcquire(200, TimeUnit.MILLISECONDS);
```

Comment:
> we can't wait long since this is always done on the cluster state update thread

Should we simply not wait at all here? I assume that if the semaphore is blocked for longer than 200ms then the system will continue to try in a later update? I think it would be best to avoid any waiting on the cluster state update threads all together unless there's a very good reason for it.